asciidoc/courses/llm-fundamentals/modules/1-introduction/lessons/1-neo4j-and-genai/lesson.adoc

```diff
@@ -0,0 +1,76 @@
+
+So we start on the same page, let's first cover the basics.
+
+== What is Neo4j?
+
+Neo4j is a graph database management system that allows us to store, manage, and query highly connected data.
+Unlike traditional relational databases, which use tables and rows, Neo4j uses a graph-based model with nodes and relationships.
+This makes it particularly well-suited for representing and querying complex, interconnected data.
+
+
+== What are Knowledge Graphs?
+
+Knowledge Graphs are a specific implementation of a Graph Database, where the underlying data is designed to capture and represent knowledge with a rich semantic context.
+They not only capture the entities and connections between, but also the meaning behind them.
+
+Knowledge graphs are tailored for semantic search, data retrieval, reasoning, and data amalgamation, frequently powering search engines, AI, and research.
+On the other hand, Neo4j's regular graphs serve diverse roles, such as social networking, recommendations, and fraud detection, not always demanding semantic depth.
+
+Knowledge graphs typically lean on ontologies, offering structured concepts within a domain and their interrelations. Conversely, while Neo4j can engage with ontologies and offers schema constraints, its standard graphs may not consistently utilize or demand structured semantic frameworks.
+
+With their semantic foundation, knowledge graphs can readily adapt and evolve as they assimilate new information and face domain shifts. In contrast, Neo4j, with its schema-free design, emphasizes flexibility in data accommodation rather than adjustments based on changing knowledge or semantics.
+
+
+== Generative AI & Large Language Models
+
+Generative AI refers to a class of algorithms and models that can generate new content, such as images, text, or even music, based on patterns and examples from existing data. These models learn the underlying structure and distribution of the data and can then generate new samples that resemble the original data.
+
+Generative AI has found applications in a wide range of fields, including image synthesis, text generation, chatbots, and recommendation systems. By leveraging the power of machine learning and deep learning techniques, generative AI models can generate highly realistic and creative outputs.
+
+Large Language Models, most commonly referred to as LLMs, are machine learning models trained on vast amounts of text data to understand and generate human-like text. They can answer questions, create content, and assist with various linguistic tasks by leveraging patterns learned from the data.
+
+
+== Potential Problems
+
+While LLMs provide a lot of potential, they should also be used with caution.
+
+At their core, LLMs are trained to predict the next word(s) in a sequence.
+They convert words into tokens, text into complex vector embeddings; arrays of numbers that can be compared and contrasted.
```

Comment:
I would change this paragraph, focus on the sequence prediction, and that they are using their pretrained data (which includes a lot of questionable sources) + are fine-tuned to be helpful at all costs - that creates the hallucination.

With the internal representation they have a concept of the meaning/essence of words or phrases but no ethical or intelligent understanding.